02_LabJat/bubble_sort_test.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInRageOutputBubbleSort(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := "[1 2 3 5]\n"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestEmptyArrayBubbleSort(t *testing.T) {
+	// Given
+	r := require.New(t)
+	// When
+	outPut := bubbleSort([]int{})
+
+	// Then
+
+	actual := 0
+	r.Equalf(len(outPut), actual, "Unexpected output in main()")
+}
+func TestSingleElementArrayBubbleSort(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	outPut := bubbleSort([]int{1})
+
+	// Then
+
+	actual := 1
+	r.Equalf(len(outPut), actual, "Unexpected output in main()")
```

Comment:
Same as above.
Also compare the actual output with [ElementsMatch](https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch):
```go
r.ElementsMatch([]int{}, outPut)
```

Reply:
Updated the PR using elements Match